src/State.hs

```diff
@@ -1286,6 +1286,31 @@ deleteMessage new = do
   cId <- use csCurrentChannelId
   when (postChannelId new == cId) updateViewed
 
+maybePostUsername :: ChatState -> Post -> T.Text
+maybePostUsername st p = case do { uId <- postUserId p; name <- usernameForUserId uId st; return name } of
+    Nothing -> T.empty
+    Just x -> x
+
+maybeNotify :: [PostToAdd] -> MH ()
+maybeNotify [] = do 
+    return ()
+maybeNotify (OldPost _:ps) =
+    maybeNotify ps
+maybeNotify (RecentPost p m:ps) = do
```

Comment:
Can you factor the body of this case out as its own function? That will help clarify what is being done here.

Reply:
Also, can you use more descriptive names than `m` and `p` here?